Recursively zip results folder using relative path

hnpl opened the pull request:
Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>

hnpl committed:
Recursively zip results folder using relative path

Signed-off-by: Hoa Nguyen <hoanguyen@ucdavis.edu>

powerjg commented on run/gem5art/run.py:
```diff
@@ -472,8 +472,11 @@ def saveResults(self) -> None:
+            for root, dirs, files in os.walk(self.relative_outdir):
+                if root == self.relative_outdir:
+                    files = filter(lambda f: f != 'results.zip', files)
+                for f in fiels:
```
typo! Someone didn't test this code ;)

powerjg commented on run/gem5art/run.py:
```diff
@@ -472,8 +472,11 @@ def saveResults(self) -> None:
-            for f in files:
-                zipf.write(os.path.join(self.outdir, f))
+            for root, dirs, files in os.walk(self.relative_outdir):
+                if root == self.relative_outdir:
```
This won't merge cleanly with the changeset removing the relative outdir... What will need to be done to fix this?

hnpl replied:
So when we create the gem5Run object, we must give `outdir` an abs path?

powerjg replied:
Yes. See https://github.com/darchr/gem5art/pull/50 and https://github.com/darchr/gem5art/pull/50/commits/9a139d613e8c11ab4ace0647ec26d75d05607a13.

I don't think there's any way around that. However, there will be a `cwd` in run (https://github.com/darchr/gem5art/pull/50/commits/9a139d613e8c11ab4ace0647ec26d75d05607a13#diff-7df16e1be50255dbdcbdaefa049d0cc3R379) if you wanted to use a directory relative to "results".

However, it's probably better not to zip based on the "results" directory, but to just zip the files in the output. So, if the output dir is `/home/jlp/gem5/experiment/results/linux/X86/cooltest/config1/` I would simply zip relative to `config1/`

hnpl replied:
So we can find the first subdirectory of that `cwd` that has a file, and zip everything relative to that folder?

hnpl replied:
Oh, we can simply look for the subdirectory containing simout / simerr files and zip everything relative to that folder.

powerjg reviewed (commented):
I can't find my other comment to delete it. Please ignore if you see it.

powerjg commented:
Yeah. That sounds right. However, you don't have to search. That directory
is outdir.



On Mon, Dec 9, 2019, 3:59 PM Hoa Nguyen <notifications@github.com> wrote:

> *@takekoputa* commented on this pull request.
> ------------------------------
>
> In run/gem5art/run.py
> <https://github.com/darchr/gem5art/pull/53#discussion_r355750820>:
>
> > @@ -472,8 +472,11 @@ def saveResults(self) -> None:
>
>          with zipfile.ZipFile(os.path.join(self.outdir, 'results.zip'), 'w',
>                               zipfile.ZIP_DEFLATED) as zipf:
> -            for f in files:
> -                zipf.write(os.path.join(self.outdir, f))
> +            for root, dirs, files in os.walk(self.relative_outdir):
> +                if root == self.relative_outdir:
>
> Oh, we can simply look for the subdirectory containing simout / simerr
> files and zip everything relative to that folder.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/darchr/gem5art/pull/53?email_source=notifications&email_token=AAA2YHFT2VF7HLO3HTO46ELQX3LXFA5CNFSM4JYUINTKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCOR4WGY#discussion_r355750820>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA2YHCCS23VQ6J6JUL7XA3QX3LXFANCNFSM4JYUINTA>
> .
>

powerjg commented:
What's the status of this? No rush, but I'd like to include this change in the next release as well.

aakahlow commented:
I am going to look at this as Hoa is going to be busy with his class stuff today

aakahlow committed:
update the naming of zipped files

Now the files will be recursively zipped in results folder
and the their names will relative to the last folder of outdir
e.g. 401.bzip2/stats.txt, 401.bzip2/speclog/sth (The change assumes
that self.outdir is an absolute path)

aakahlow committed:
fix the type checking error

powerjg reviewed (approved):
LGTM, assuming you've tested it :)
